pkg/secrets/repositoryserver/aws_secrets_test.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2023 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package repositoryserver
+
+import (
+	"github.com/pkg/errors"
+	. "gopkg.in/check.v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	secerrors "github.com/kanisterio/kanister/pkg/secrets/errors"
+)
+
+type AWSSecretCredsSuite struct{}
+
+var _ = Suite(&AWSSecretCredsSuite{})
+
+func (s *AWSSecretCredsSuite) TestValidateRepoServerAWSCredentials(c *C) {
+	for _, tc := range []struct {
+		secret       Secret
+		errorChecker func(err error)
+	}{
+		{ // Valid AWS Secret
+			secret: NewAWSLocation(&v1.Secret{
+				Type: v1.SecretType(LocTypeS3),
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "sec",
+					Namespace: "ns",
+				},
+				Data: map[string][]byte{
+					BucketKey: []byte("bucket"),
+					RegionKey: []byte("region"),
+				},
+			}),
+			errorChecker: func(err error) {
+				c.Check(err, IsNil)
+			},
```

Comment:
Instead of using `func()` as errorChecker and validating it against all the test cases. Should we use `expectedError` and `expectedErrorMsg` separate fields and validate those? It will reduce number of line (as we are repeating `c.Check()` for each test case) as well as improve readability